### PR TITLE
Show skip link properly with grace period

### PR DESCRIPTION
**Pull request by whaught**
## Proposed Changes

* Ensures that the 'skip' option is shown for MFARequired realms if the user is within the grace-period.

### CODE REVIEW

File: pkg/database/realm.go [hunk 1]
```diff
@@ -207,6 +207,19 @@ type Realm struct {
 	Tokens []*Token            `gorm:"PRELOAD:false; SAVE_ASSOCIATIONS:false; ASSOCIATION_AUTOUPDATE:false, ASSOCIATION_SAVE_REFERENCE:false"`
 }
 
+func (realm *Realm) EffectiveMFAMode(user *User) *AuthRequirement {
```
Comment (jeremyfaller) [inline pkg/database/realm.go:210]: return by value, saves a bunch of this mucking about with temps.

Comment (jeremyfaller) [reply pkg/database/realm.go:210]: Also saves the indirection in the caller.

Comment (whaught) [reply pkg/database/realm.go]: lol idk how I got so far with these const -> var pointers without thinking "this is easier as a value". Thanks

File: pkg/controller/login/register.go [hunk 2]
```diff
@@ -31,8 +31,18 @@ func (c *Controller) HandleRegisterPhone() http.Handler {
 			return
 		}
 
+		user := controller.UserFromContext(ctx)
```
Comment (sethvargo) [inline pkg/controller/login/register.go:34]: Nit: `currentUser`

File: pkg/database/realm.go [hunk 3]
```diff
@@ -207,6 +207,17 @@ type Realm struct {
 	Tokens []*Token            `gorm:"PRELOAD:false; SAVE_ASSOCIATIONS:false; ASSOCIATION_AUTOUPDATE:false, ASSOCIATION_SAVE_REFERENCE:false"`
 }
 
+func (r *Realm) EffectiveMFAMode(user *User) AuthRequirement {
```
Comment (sethvargo) [inline pkg/database/realm.go:210]: Needs docs since the value changes depending on time.

**Review by sethvargo — CHANGES_REQUESTED** [review]

**Review by sethvargo — APPROVED** [review]
/lgtm

**Pull request merged**